src/server/frontend_wayland/xdg_shell_stable.cpp

```diff
@@ -200,7 +200,14 @@ mf::XdgSurfaceStable::XdgSurfaceStable(wl_resource* new_resource, WlSurface* sur
 
 void mf::XdgSurfaceStable::get_toplevel(wl_resource* new_toplevel)
 {
-    auto toplevel = new XdgToplevelStable{new_toplevel, this, surface};
+    if (!surface)
+    {
+        BOOST_THROW_EXCEPTION(mw::ProtocolError(
+            resource,
+            mw::generic_error_code,
```

Comment:
Maybe this (and all the other ones) are [defunct_role_object](https://wayland.app/protocols/xdg-shell#xdg_surface:enum:error:entry:defunct_role_object)?

Reply:
We can't use that yet because we haven't bumped to XDG shell stable v5. Event when we do I don't think that's right since that's an XDG surface error and we're sending the error on the XDG toplevel (because the XDG surface doesn't exist).